bgpd/bgp_routemap.c

```diff
@@ -4720,6 +4720,23 @@ static void bgp_route_map_delete(const char *rmap_name)
 	route_map_notify_dependencies(rmap_name, RMAP_EVENT_MATCH_DELETED);
 }
 
+bool bgp_route_map_has_extcommunity_rt(const struct route_map *map)
+{
+	struct route_map_index *index = NULL;
+	struct route_map_rule *set = NULL;
+
+	assert(map);
+
+	for (index = map->head; index; index = index->next) {
+		for (set = index->set_list.head; set; set = set->next) {
+			if (set->cmd && set->cmd->str &&
+			    strmatch(set->cmd->str, "extcommunity rt"))
```

Comment:
Not sure I understand this logic here. Don't say it's wrong, but in the long-term, we should have something like a custom enum for distinguishing a specific command for `struct routemap_hook_context`.

Reply:
Today, the framework is only available for delete.
And the framework does not distinguish between create and modify events.

that is why I chose this method.

Reply:
I thought this would allow the operator to change the RT, not just delete it, right?